server/src/main/java/org/apache/kafka/server/RPCProducerIdManager.java

```diff
@@ -0,0 +1,191 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.server;
```

Comment:
Maybe this class should be moved to `transaction-coordinator` module?

Reply:
I think it can't move to `transaction-coordinator` module, there is a dependency circle.
```
:coordinator-common:compileJava
\--- :storage:compileJava
     \--- :transaction-coordinator:compileJava
          \--- :server:compileJava
               +--- :group-coordinator:compileJava
               |    +--- :coordinator-common:compileJava (*)
               |    \--- :storage:compileJava (*)
               +--- :storage:compileJava (*)
               \--- :transaction-coordinator:compileJava (*)
```
In `transaction-coordinator` modules, we need add dependency like this
```
  dependencies {
    implementation libs.jacksonDatabind
    implementation project(':clients')
    implementation project(':server')
    implementation project(':server-common')
    implementation libs.slf4jApi

    testImplementation libs.junitJupiter
    testImplementation libs.mockitoCore
    testRuntimeOnly runtimeTestLibs
    testImplementation project(':clients').sourceSets.test.output
    testImplementation project(':test-common')
    testImplementation project(':test-common:test-common-api')

    generator project(':generator')
  }
  ```

Reply:
why `transaction-coordinator` needs to depend on `server` module if we move both `ProducerIdManager` and `RPCProducerIdManager` to `transaction-coordinator`

Reply:
in `ProducerIdManager` we need to use `org.apache.kafka.server.NodeToControllerChannelManager` in this method, thus it need server module
```
    static ProducerIdManager rpc(int brokerId,
                                 Time time,
                                 Supplier<Long> brokerEpochSupplier,
                                 NodeToControllerChannelManager controllerChannel) {
        return new RPCProducerIdManager(brokerId, time, brokerEpochSupplier, controllerChannel);
    }
```

Reply:
It seems to me `NodeToControllerChannelManager`  should be in server-common module as it can be used by other server-related module

Reply:
Should I also move it to server-common module in this PR ?

Reply:
yep